### PR TITLE
feat(CX-3675): Delete Artist Mutation

### PR DESCRIPTION
**Pull request by lordkiz**
Partly resolves [CX-3675]
### Description
Adds the deleteArtist mutation to Metaphysics
Relates to https://github.com/artsy/gravity/pull/16403

[CX-3675]: https://artsyproduct.atlassian.net/browse/CX-3675?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/schema/v2/artist/deleteArtistMutation.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      resolve: (result) => result,
+    },
+  },
+  mutateAndGetPayload: ({ id }, { deleteArtistLoader }) => {
+    if (!deleteArtistLoader) {
+      throw new Error("You need to be signed in to perform this action")
+    }
+
+    return deleteArtistLoader(id)
+      .then((result) => result)
```
Comment (olerichter00) [inline src/schema/v2/artist/deleteArtistMutation.ts:67]: I think we can remove this line.

```suggestion
```

Comment (lordkiz) [reply src/schema/v2/artist/deleteArtistMutation.ts:67]: 👀 The success type resolves artist. Unless you mean I should refactor to async func?

Comment (olerichter00) [reply src/schema/v2/artist/deleteArtistMutation.ts:67]: `deleteArtistLoader(id)` already returning the same. Appending `.then((result) => result)` isn't doing anything.

Comment (lordkiz) [reply src/schema/v2/artist/deleteArtistMutation.ts:67]: Ahhh. I did not initially realise what you mean. Sure. Will update. 👍

File: src/schema/v2/artist/deleteArtistMutation.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import { GraphQLNonNull, GraphQLString } from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import {
+  formatGravityError,
+  GravityMutationErrorType,
+} from "lib/gravityErrorHandler"
+import { ResolverContext } from "types/graphql"
+import { GraphQLObjectType } from "graphql"
+import { ArtistType } from "./index"
+import { GraphQLUnionType } from "graphql"
```
Comment (MrSltun) [inline src/schema/v2/artist/deleteArtistMutation.ts:10]: **Nitpick:** Maybe we should put the GraphQL types in one line 👀 

```suggestion
import { GraphQLNonNull, GraphQLObjectType, GraphQLString, GraphQLUnionType } from "graphql"
import { mutationWithClientMutationId } from "graphql-relay"
import {
  formatGravityError,
  GravityMutationErrorType,
} from "lib/gravityErrorHandler"
import { ResolverContext } from "types/graphql"
import { ArtistType } from "./index"
```

File: src/schema/v2/artist/__tests__/deleteArtistMutation.test.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  })
+
+  it("throws error when data loader is missing", async () => {
+    const errorResponse = "You need to be signed in to perform this action"
+
+    try {
+      await runQuery(mutation)
+      throw new Error("An error was not thrown but was expected.")
+    } catch (error) {
+      // eslint-disable-next-line jest/no-conditional-expect, jest/no-try-expect
```
Comment (MrSltun) [inline src/schema/v2/artist/__tests__/deleteArtistMutation.test.ts:58]: 👀

**Review by MrSltun — APPROVED** [review]
Looks great! 🙌  I just left a few comments
I'm also a bit scared of moving forward quickly with this, should we add a check if the artist is a personal artist and belongs to the user before running the mutation?

**Comment by lordkiz** [comment]
> I'm also a bit scared of moving forward quickly with this, should we add a check if the artist is a personal artist and belongs to the user before running the mutation?

@MrSltun Thanks. These checks are already in place in the backend.

**Comment by MrSltun** [comment]
@lordkiz I saw the PR for adding the checks, but maybe we need to test it against the endpoint alongside the module?

**Review by olerichter00 — APPROVED** [review]

**Review by MounirDhahri — DISMISSED** [review]

**Review by MrSltun — DISMISSED** [review]

**Pull request merged**